ConcordiumWallet/Views/Stake/Baking/ComissionSettings/BakerCommissionSettingsViewModel.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  BakerCommissionSettingsViewModel.swift
+//  ConcordiumWallet
+//
+//  Created by Milan Sawicki on 06/11/2023.
+//  Copyright © 2023 concordium. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+extension NumberFormatter {
+    static var commissionFormatter: NumberFormatter {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        formatter.multiplier = 100
+        formatter.maximumFractionDigits = 3
+        return formatter
+    }
+}
+
+enum BakerCommissionSettingError: LocalizedError {
+    case transactionFeeOutOfRange
+    case bakingRewardOutOfRange
+    case finalizationRewardOutOfRange
+    case networkError(Error)
+
+    var errorMessage: String {
+        switch self {
+        case .bakingRewardOutOfRange:
+            return "Baking reward is out of specified range"
+        case .finalizationRewardOutOfRange:
+            return "Finalization reward is out of specified range"
+        case .transactionFeeOutOfRange:
+            return "Transaction fee is out of specified range"
+        case let .networkError(error):
+            return error.localizedDescription
+        }
+    }
+}
+
+class BakerCommissionSettingsViewModel: ObservableObject {
+    @Published var transactionFeeCommission: Double = 0
+    @Published var finalizationRewardCommission: Double = 0
+    @Published var bakingRewardCommission: Double = 0
+    @Published var commissionRanges: (
+        bakingCommissionRange: CommissionRange,
+        transactionCommissionRange: CommissionRange,
+        finalizationCommissionRange: CommissionRange
+    )?
+    @Published var error: BakerCommissionSettingError?
+    var dismissView: () -> Void
+    private var cancellables = Set<AnyCancellable>()
+    private var didTapContinue: () -> Void
+    private var service: StakeServiceProtocol
+    private var handler: StakeDataHandler
+
+    init(
+        service: StakeServiceProtocol,
+        handler: StakeDataHandler,
+        didTapContinue: @escaping (() -> Void),
+        dismissView: @escaping (() -> Void)
+    ) {
+        self.service = service
+        self.didTapContinue = didTapContinue
+        self.handler = handler
+        self.dismissView = dismissView
+    }
+
+    func fetchData() {
+        service.getChainParameters().asResult().sink { result in
+            switch result {
+            case let .success(response):
+
+                self.commissionRanges = (bakingCommissionRange: response.bakingCommissionRange,
+                                         transactionCommissionRange: response.transactionCommissionRange,
+                                         finalizationCommissionRange: response.finalizationCommissionRange
+                )
```

Comment:
```suggestion
                self.commissionRanges = (
                    bakingCommissionRange: response.bakingCommissionRange,
                    transactionCommissionRange: response.transactionCommissionRange,
                    finalizationCommissionRange: response.finalizationCommissionRange
                )
```
Isn't that the style we usually prefer?